### PR TITLE
Asynchronous character assets view

### PR DESCRIPTION
**Pull request by herpaderpaldent**
Asynchronous character assets view
* Reuse of warlof's attempt.
* impossible to use DataTable because of the location segmentation
* asynchronous load of tablerows
* box overlay during load.
* This resolves

This solves: https://github.com/eveseat/seat/issues/417 and could potentially be a way for: https://github.com/eveseat/seat/issues/352 

depends on eveseat/services#66

### CODE REVIEW

**Comment by leonjza** [comment]
This should really just mimic the characters assets view logic.

**Comment by herpaderpaldent** [comment]
I moved the rendering to the ajax request because the linked issue regards user with lots of assets. If the rendering happens on the page load, the controller can run into a timout. That is why i moved it.

Apart from this i suggested on slack that we could start splitting the assets in multiple tables -> per location an own table (made with server-side-datatables). What do you think of this?

**Comment by warlof** [comment]
It will of course as it's the server which has the final word. The controller himself is only routing stuff and returning rendered page.

Here, you'll get the page shown, but in case of heavy load, the container will remain empty.

**Comment by herpaderpaldent** [comment]
I tried to implement a paginated ajax call to avoid this.  But because of the table design with the row-span for each individual location this could have resulted f.e. in multiple Jita-Locations. 

The only other way i now see is either accept the multiple location tab (whilst increading the paginated elements count to an amount to assume that it wouldn't happen) or split locations in individual tables. 

What whould you do?

**Comment by warlof** [comment]
The datatable thing is maybe something to dig a bit into.

Just keep in mind that in order to reduce the load, the thing is to load content of an item using an ajax call.
And not using a one shoot ajax call to load the entire tree.

**Comment by herpaderpaldent** [comment]
This is now a full server-side datatable. Something i added too:
* Linker Character Assets can be viewed in a 2nd Tab. It is fully searchable by the item name and the custom name you might have given the item:
image

File: src/Http/Controllers/Character/AssetsController.php [hunk 1]
```diff
@@ -55,8 +54,74 @@ public function getAssets(int $character_id)
...
+    }
+
+    /**
+     * @param int $character_id
+     *
+     * @return mixed
+     */
+    public function getCharacterAssets(int $character_id)
+    {
+        if(request('all_linked_characters') === 'false')
```
[the remark this answers is not in the document]
Comment (herpaderpaldent) [reply src/Http/Controllers/Character/AssetsController.php]: i can't reproduce this finding: 
1. After calling the `Controller::getAssetView()` method, the param `all_linked_characters` is set: https://github.com/eveseat/web/pull/209/files#diff-112b9ff23aa7b49376ac927177422962R68 
2. in this `getCharacterAssets(int $character_id)` is set based on the URL param from route:
https://github.com/eveseat/web/pull/209/files#diff-e9958ea6f787c1535bf0455634f7dbacR45

i take this finding might be a leftover from the missing `Asset Safety` exception which i fixed with the latest commit: https://github.com/eveseat/services/pull/66/commits/3fcca9c1eb12ddb9f00c52682ed544831caa27c2

File: src/resources/views/character/assets.blade.php [hunk 2]
```diff
@@ -164,55 +42,154 @@ function($value){
 
   <script type="text/javascript">
 
-    $(".viewcontent").on("click", function () {
+    var url = "{{route('character.view.assets.details', ['character_id' => auth()->user()->id])}}";
```
[the remark this answers is not in the document]
Comment (herpaderpaldent) [reply src/resources/views/character/assets.blade.php]: You are absolutely right and i changed the implementation accordingly: https://github.com/eveseat/web/pull/209/commits/942c3da04b948e44c567d1d41ccd268d81051ccf

**Comment by herpaderpaldent** [comment]
Note to myself: i need to test this with assets in delivery too

**Comment by herpaderpaldent** [comment]
added translations just now

**Pull request merged**